### PR TITLE
Input validation

### PR DESCRIPTION
**Pull request by jchung722**
Input validation for spots that are already taken
**Also reformatted the tests to make board data structure more readable. Should have probably been a  separate pull request...sorry D:

### CODE REVIEW

File: test/input_test.exs [hunk 1]
```diff
@@ -2,19 +2,42 @@ defmodule InputTest do
   use ExUnit.Case
   doctest Input
 
-  describe "when input is valid" do
-    test "user input is returned as integer" do
-      assert Input.process("1\n") == {:ok, 1}
+  describe "input is valid" do
+    test "and user input is parsed and returned as index of the board" do
```
Comment (nystromb) [inline test/input_test.exs]: I feel like this test shouldn't really be under this describe block. I would expect any test under the "input is valid" block to be only calling the `validate` method. Perhaps this could be moved outside of this describe block? 

Also, just kind of a side note about test descriptions. Usually when I see the words "and" or "or" used in a test description, that tells me that this test is probably testing multiple things. But that's not the case here. So I might just shorten this test description to be `returns the board index from user input`

File: test/input_test.exs [hunk 2]
```diff
@@ -2,19 +2,42 @@ defmodule InputTest do
...
+      assert Input.validate("invalid\n", board) == :invalid
     end
+
+    test "the spot on the board is already taken" do
+      board = ["1", "X", "3",
+               "4", "O", "6",
+               "7", "8", "9"]
+      assert Input.validate("2", board) == :invalid
+    end
+
```
Comment (nystromb) [inline test/input_test.exs:41]: What happens when you enter a number that is outside of the range of the board, like 10 for example?

File: test/status_test.exs [hunk 3]
```diff
@@ -21,68 +23,98 @@ defmodule StatusTest do
 
   describe "during game" do
     test "a game with no three adjacent moves from a player has no winner" do
-      board = ["X", "2", "X", "O", "5", "6", "O", "8", "9"]
+      board = ["X", "2", "X",
+               "O", "5", "6",
```
Comment (nystromb) [inline test/status_test.exs:27]: 🔥 Really like this change, makes it a bit easier for me to read!!

File: lib/input.ex [hunk 4]
```diff
@@ -1,13 +1,31 @@
...
+    else
+      :invalid -> :invalid
+      :error -> :invalid
     end
   end
+
+  defp spot_check(input, board) do
+    input_board_index = Input.to_board_index(input)
+    cond do
+      Enum.at(board, input_board_index) == "X" ->
```
Comment (nystromb) [inline lib/input.ex]: Can this be moved above and assigned to a variable? It might make this code a bit more explicit

**Review by nystromb — COMMENTED** [review]
Looking good! Just some questions

**Comment by nystromb** [comment]
BTW, I like having that refactor just as a commit instead of an entire PR. Since its a small refactor I kind of like baking that work into future stories as needed 👍

File: test/input_test.exs (hunk 2 above)
Comment (jchung722) [reply test/input_test.exs:41]: Oh yes, the program doesn't crash or update the board, but currently it does switch the player as if it's a valid input. Will be updating validation for this in a future story!

Comment (nystromb) [reply test/input_test.exs:41]: Is there a reason we can't handle that in this PR? I feel that the processing of the move that is outside of the range the board supports is a bug and essentially the last edge case for this story

Comment (jchung722) [reply test/input_test.exs:41]: Don't mind doing that for this PR if you give are giving the okay 👌

Comment (nystromb) [reply test/input_test.exs:41]: Nevermind, actually! I should have looked at the board for the acceptance criteria. I see that these are two separate stories.

LGTM

**Review by nystromb — APPROVED** [review]
💯
